lib/fastlane/actions/xcode_server_get_assets.rb

```diff
@@ -0,0 +1,299 @@
+
+            # unzip it, it's a .tar.gz file
+            out_folder = File.join(dir, "out_#{rand(1000000)}")
+            FileUtils.mkdir_p(out_folder)
+
+            action.sh "cd \"#{out_folder}\"; cat \"#{temp_file}\" | gzip -d | tar -x"
+
+            # then pull the real name from headers
+            asset_filename = response.headers['Content-Disposition'].split(';')[1].split('=')[1].gsub('"', '')
+            asset_foldername = asset_filename.split('.')[0]
+
+            # rename the folder in out_folder to asset_foldername
+            found_folder = Dir.entries(out_folder).select { |item| item != '.' && item != '..' }[0]
+
+            raise "Internal error, couldn't find unzipped folder".red if found_folder == nil
+
+            unzipped_folder_temp_name = File.join(out_folder, found_folder)
+            unzipped_folder = File.join(out_folder, asset_foldername)
+
+            # rename to destination name
+            FileUtils.mv(unzipped_folder_temp_name, unzipped_folder)
+
+            target_folder = File.absolute_path(target_folder)
+
+            # create target folder if it doesn't exist
+            FileUtils.mkdir_p(target_folder)
+
+            # and move+rename it to the destination place
+            FileUtils.cp_r(unzipped_folder, target_folder)
+            out = File.join(target_folder, asset_foldername)
+            return out
+          end
+          return nil
+        end
+
+        def headers
+          require 'base64'
+          headers = { 
+            'User-Agent' => 'fastlane-xcode_server_get_assets', #XCS wants user agent. for some API calls. not for others. sigh.
+            'X-XCSAPIVersion' => 1 #using XCS API version 3, Xcode needs this otherwise it explodes in a 500 error fire
+          }
+
+          if @username && @password 
+            userpass = "#{@username}:#{@password}"
+            headers['Authorization'] = "Basic #{Base64.strict_encode64(userpass)}"
+          end
+          
+          return headers
+        end
+
+        def get_endpoint(endpoint, response_block=nil)
+          url = url_for_endpoint(endpoint)
+          headers = self.headers || {}
+
+          if response_block
+            response = Excon.get(url, :response_block => response_block, :headers => headers)
+          else
+            response = Excon.get(url, :headers => headers)
+          end
+
+          return response
+        end
+
+        private 
+
+        def url_for_endpoint(endpoint)
+          "#{@host}:20343/api#{endpoint}"
+        end
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Downloads Xcode Bot assets like the `.xcarchive` and logs"
+      end
+
+      def self.details
+        "This action downloads assets from your Xcode Server Bot (works with Xcode Server
```

Comment:
Yeah it looks alright I think https://www.dropbox.com/s/1ke7ho2weca8il8/Screenshot%202015-08-11%2016.20.23.png?dl=0